Sources/FirebaseFirestore/vendor/Codable/DocumentID.swift

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+extension CodingUserInfoKey {
+  static let documentRefUserInfoKey =
```

Comment:
I would just use a `static var` here with a computed value

Reply:
Sure

Reply:
Actually, since this a vendored file from upstream I'm going to decline this change since this is exactly how it is in the matching SDK. I don't see a huge upshot of making this conversion from what the authors intended.